picard/util/filenaming.py

```diff
@@ -527,6 +530,39 @@ def make_save_path(path, win_compat=False, mac_compat=False):
     return path
 
 
+WINDOWS_FORBIDDEN_NAMES = {
+    'CON', 'PRN', 'AUX', 'NUL',
+    *('COM%d' % i for i in range(1, 10)),
+    'COM¹', 'COM²', 'COM³',
+    *('LPT%d' % i for i in range(1, 10)),
+    'LPT¹', 'LPT²', 'LPT³',
+}
+
+WINDOWS_FORBIDDEN_NAMES_RE = re.compile(
+    r'(^|[%s])(%s)(?=$|[\.%s])' % (
+        re.escape(os.path.sep + (os.altsep if os.altsep else '')),
+        '|'.join(re.escape(name) for name in WINDOWS_FORBIDDEN_NAMES),
+        re.escape(os.path.sep + (os.altsep if os.altsep else '')),
+    ),
+    re.IGNORECASE
+)
```

Comment:
Really minor point, but I wonder if the duplicate code could be avoided by using `.format()`, something like (untested):

```python
WINDOWS_FORBIDDEN_NAMES_RE = re.compile(
    r'(^|[{separators}])({names})(?=$|[\.{separators}])'.format(
        separators=re.escape(os.path.sep + (os.altsep if os.altsep else '')),
        names='|'.join(re.escape(name) for name in WINDOWS_FORBIDDEN_NAMES),
    ),
    re.IGNORECASE
)
```